### PR TITLE
Permissive/Strict Options

### PR DESCRIPTION
**Pull request by njhale**
- Adds a permissive option to the intializer and configmap-server commands that, when given, permits load errors with out failing.
- Adds a strict option to the appregistry-server command that, when given, fails when load errors are encountered.

### CODE REVIEW

File: cmd/configmap-server/main.go [hunk 1]
```diff
@@ -96,26 +103,34 @@ func runCmdFunc(cmd *cobra.Command, args []string) error {
...
+	}
+	if store == nil {
+		store = registry.NewEmptyQuerier()
 	}
 
 	// sanity check that the db is available
 	tables, err := store.ListTables(context.TODO())
 	if err != nil {
-		logger.Fatalf("couldn't list tables in db, incorrect config: %v", err)
+		logger.WithError(err).Warnf("couldn't list tables in db")
```
Comment (jpeeler) [inline cmd/configmap-server/main.go]: Is there a chance of success at this point?

Comment (njhale) [reply cmd/configmap-server/main.go]: At this point, the store is effectively empty. I added this to keep in line with our theme of being permissive where we can. I figured an empty store should be non-fatal. Do you think it makes sense for the `permissive` option to gate this branch point too?

Comment (jpeeler) [reply cmd/configmap-server/main.go]: I was going back and forth on it.

**Pull request merged**